### PR TITLE
add tts

### PR DESCRIPTION
**Pull request by DDMeaqua**
#### 💻 变更类型 | Change Type

- [x] feat    
- [ ] fix    
- [ ] refactor    
- [ ] perf    
- [ ] style    
- [ ] test    
- [ ] docs    
- [ ] ci    
- [ ] chore    
- [ ] build    

## Release Notes

- **New Features**
  - Introduced text-to-speech (TTS) functionality, allowing users to listen to chat messages.
  - Added new speech methods across various API classes to support future speech processing capabilities.
  - Enhanced TTS configurations with default settings for voice models and speeds.
  - Implemented new icons for speech actions in the chat component, improving user interaction.

- **Bug Fixes**
  - Enhanced error handling for speech requests.

These updates significantly enhance user interaction and accessibility within the application.

### CODE REVIEW

File: app/client/api.ts [hunk 1]
```diff
@@ -205,13 +216,16 @@ export function validString(x: string): boolean {
   return x?.length > 0;
 }
 
-export function getHeaders() {
+export function getHeaders(ignoreHeaders?: boolean) {
```
Comment (Dogtiti) [inline app/client/api.ts:219]: default `false`

File: app/client/platforms/openai.ts [hunk 2]
```diff
@@ -78,7 +79,7 @@ export interface DalleRequestPayload {
 export class ChatGPTApi implements LLMApi {
   private disableListModels = true;
 
-  path(path: string): string {
+  path(path: string, model?: string): string {
```
Comment (Dogtiti) [inline app/client/platforms/openai.ts:82]: 这里model的意义是？好像没用到？

**Review by Dogtiti — CHANGES_REQUESTED** [review]

File: app/client/platforms/openai.ts (hunk 2 above)
Comment (DDMeaqua) [reply app/client/platforms/openai.ts:82]: 下面 speech 里用到了

**Review by Dogtiti — APPROVED** [review]

File: app/client/platforms/openai.ts (hunk 2 above)
Comment (Dogtiti) [reply app/client/platforms/openai.ts:82]: path这个函数里面好像没用到吧？买看到呀

**Review by Dogtiti — CHANGES_REQUESTED** [review]

**Review by Dogtiti — APPROVED** [review]

**Pull request merged**